.travis.yml

```diff
@@ -1,76 +1,119 @@
+    - ../configure --cache-file=config.cache $BITCOIN_CONFIG_ALL $BITCOIN_CONFIG || ( cat config.log && false )
     - make distdir VERSION=$HOST
-    - cd bitcore-$HOST
-    - ./configure --cache-file=../config.cache $BITCORE_CONFIG_ALL $BITCORE_CONFIG || ( cat config.log && false)
-    - make $MAKEJOBS $GOAL || ( echo "Build failure. Verbose build follows." && make $GOAL V=1 ; false )
+    - cd "bitcore-$HOST" || (echo "could not enter distdir bitcore-$HOST"; exit 1)
+    - ./configure --cache-file=../config.cache $BITCOIN_CONFIG_ALL $BITCOIN_CONFIG || ( cat config.log && false )
+    - make $MAKEJOBS $GOAL || ( "Build failure. Verbose build follows." && make $GOAL V=1 ; false )
     - export LD_LIBRARY_PATH=$TRAVIS_BUILD_DIR/depends/$HOST/lib
-    - if [ "$RUN_TESTS" = "true" ]; then travis_wait 30 make $MAKEJOBS check VERBOSE=1; fi
-    - if [ "$TRAVIS_EVENT_TYPE" = "cron" ]; then extended="--extended --exclude pruning,dbcrash"; fi
-    - if [ "$RUN_TESTS" = "true" ]; then test/functional/test_runner.py --coverage --quiet ${extended}; fi
+    - if [ "$RUN_TESTS" = "true" ]; then travis_wait 50 make $MAKEJOBS check VERBOSE=1; fi
+    - if [ "$TRAVIS_EVENT_TYPE" = "cron" ]; then extended="--extended --exclude feature_pruning,feature_dbcrash"; fi
+    - if [ "$RUN_TESTS" = "true" ]; then test/functional/test_runner.py --combinedlogslen=4000 --coverage --quiet --failfast ${extended}; fi
+# ARM    
+    - if [ "$NAME" = "arm-linux-gnueabihf" ]; then cd $OUTDIR; tar -czf bitcore-$TRAVIS_TAG-$NAME.tar.gz -C $OUTDIR bin lib; fi
+# Win32
+    - if [ "$NAME" = "win32" ]; then mkdir -p $OUTDIR; zip -r bitcore-$NAME-$TRAVIS_TAG-$NAME.zip ./src/*.exe; cp bitcore-$NAME-$TRAVIS_TAG-$NAME.zip $OUTDIR; fi
+# 32-bit + dash   
+    - if [ "$NAME" = "i686-pc-linux-gnu" ]; then cd $OUTDIR; tar -czf bitcore-$TRAVIS_TAG-$NAME.tar.gz -C $OUTDIR bin lib; fi
+# Qt5 & system libs
+    - if [ "$NAME" = "x86_64-linux-gnu_qt5_with-libs" ]; then cd $OUTDIR; tar -czf bitcore-$TRAVIS_TAG-$NAME.tar.gz -C $OUTDIR bin lib; fi
+# Win64
+    - if [ "$NAME" = "win64" ]; then mkdir -p $OUTDIR; zip -r bitcore-$NAME-$TRAVIS_TAG-$NAME.zip ./src/*.exe; cp bitcore-$NAME-$TRAVIS_TAG-$NAME.zip $OUTDIR; fi
+# x86_64 Linux (uses qt5 dev package instead of depends Qt to speed up build and avoid timeout)
+    - if [ "$NAME" = "x86_64-linux-gnu_qt5-dev" ]; then cd $OUTDIR; tar -czf bitcore-$TRAVIS_TAG-$NAME.tar.gz -C $OUTDIR bin lib; fi
+# x86_64 Linux, No wallet
+    - if [ "$NAME" = "x86_64-linux-gnu_no-wallet" ]; then cd $OUTDIR; tar -czf bitcore-$TRAVIS_TAG-$NAME.tar.gz -C $OUTDIR bin lib; fi
+# Cross-Mac
+    - if [ "$NAME" = "osx64" ]; then mkdir -p $OUTDIR; cp BitCore-BTX.dmg $OUTDIR/bitcore-$TRAVIS_TAG-$NAME.dmg; fi
+
+    - pwd
+    - ls ./*
+    - ls ./*/*
+    - ls ..
+    - ls $BASE_OUTDIR/**/*
 after_script:
     - echo $TRAVIS_COMMIT_RANGE
     - echo $TRAVIS_COMMIT_LOG
+deploy:
+  provider: releases
+  api_key:
+    secure: <github API pubkey> #TODO
```

Comment:
**TODO**
### Create a GitHub API Token
Go to settings->developer settings->Personal access tokens (https://github.com/settings/tokens) in the GitHub UI. 

Create a new token, removing every permission except repo.